app/Models/Campaign.php

```diff
@@ -126,6 +127,21 @@ public function scopeWhereDoesNotHaveWebsite($query)
         return $query->whereNull('contentful_campaign_id');
     }
 
+    /**
+     * Scope a query to only include campaigns containing specified cause within
+     * "cause" field.
+     */
+    public function scopeWhereHasCause($query, $cause)
+    {
+        // Sanitize the input to prevent melicious regex pattern injection:
+        $sanitizedCause = Str::slug($cause);
```

Comment:
Also just thought of this – would it be clearer (and perhaps safer) to match against the list of valid causes?

```php
$sanitizedCause = array_intersect([strtolower($cause)], Cause::all());
```